staker/staker.go

```diff
@@ -294,8 +303,22 @@ func (s *Staker) checkLatestStaked(ctx context.Context) error {
 		log.Info("latest valid not yet in our node", "staked", stakedGlobalState)
 		return nil
 	}
+
+	processedCount, err := s.txStreamer.GetProcessedMessageCount()
+	if err != nil {
+		return err
+	}
+
+	if processedCount < count {
+		log.Info("execution catching up to last validated", "validatedCount", count, "processedCount", processedCount)
+		return nil
+	}
+
 	if s.blockValidator != nil && s.config.StartFromStaked {
-		return s.blockValidator.AssumeValid(count, stakedGlobalState)
+		s.blockValidator.UpdateLatestStaked(count, stakedGlobalState)
```

Comment:
Is there a reason why we can't put the blockValidator in `s.notifiers` if `s.config.StartFromStaked`? Is that config field hot reloadable? I think it'd be fine to remove the hot reloadability if so in favor of simplicity.